lib/Controller/SettingsController.php

```diff
@@ -63,12 +63,59 @@ public function __construct(
 		$this->crypto = $crypto;
 	}
 
+	public function isDiscoveryEndpointValid($url) {
+		// Initialize cURL session
```

Comment:
This should use https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/http_client.html, not plain curl